Log4j2 integration

koder95 opened the pull request.

koder95 committed:
Update .gitignore

koder95 committed:
Update pom.xml

koder95 committed:
Create log4j2.xml

koder95 committed:
Add logging to authentication and order services

Integrated Log4j logging into Main, AuthenticationServiceImpl, and OrderServiceImpl. Added debug and info log messages to track method calls and important events such as user login and order completion, replacing previous printStackTrace usage for error handling.

mateacademy-ai-mentor commented on src/main/java/mate/academy/Main.java:
```diff
@@ -6,15 +6,19 @@
             user = authenticationService.login("bob", "1234");
         } catch (AuthenticationException e) {
-            e.printStackTrace();
+            log.error(e);
```
This violates the checklist guidance: "Let's make our log messages informative" — `log.error(e);` provides no context. Replace with a contextual message and pass the exception so the stack trace is logged, e.g. `log.error("Authentication failed for provided credentials", e);` or include the login parameter if available.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

I am requesting changes because a core checklist requirement is not met. In `src/main/java/mate/academy/Main.java`, the catch block logs only `log.error(e);` (around the catch block for authentication), which lacks context and violates the requirement to make log messages informative [CHECKLIST ITEM #3]. Please change it to include a descriptive message and pass the exception so the stack trace is preserved, for example: `log.error("Authentication failed for user {}", login, e);` or at minimum `log.error("Authentication failed", e);`.

Additionally, the task requires a Log4j2 configuration that writes to both Console and File appenders. I can’t verify `src/main/resources/log4j2.xml` in the submission. Ensure you add it with both appenders and that the File appender writes to `logs/` (e.g., `logs/app.log`) so it aligns with `.gitignore`.

Positives: your services use parameterized, informative logging and avoid `System.out.println()`/`e.printStackTrace()`. Once the Main exception log is fixed and the Log4j2 config with console+file is confirmed, this will meet the requirements. Keep it up—small tweaks and you’re there.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

koder95 committed:
Improve logging in authentication failure

Refactored the login variable for reuse and enhanced the error log message to include the username when authentication fails.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

I am requesting changes because a core requirement is not confirmed: the Log4j2 configuration file with both Console and File appenders writing to the `logs/` directory is missing or unverified. Without `src/main/resources/log4j2.xml` (or equivalent), logging may not output to console and file as required, which blocks task completion.

Critical items to fix:
- Add/verify `src/main/resources/log4j2.xml` that defines both a Console appender and a File or RollingFile appender writing to `logs/app.log` (or similar) and reference both in the `root` logger. For example, ensure attributes like `fileName="logs/app.log"` and `appenders="Console, File"` are set. This directly satisfies the requirement for console and file logging and aligns with your `.gitignore` entry for `logs/`.

What you did well:
- Application code uses Log4j2 correctly with parameterized, informative messages. No `System.out.println()` or `e.printStackTrace()` usages, and exceptions are logged with the throwable preserved.

Minor improvement (non-blocker):
- Ensure all source files end with a newline to keep consistency.

Once the logging configuration is added/confirmed, your solution should meet all requirements. Keep up the good work on the clean and contextual logging.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

koder95 commented on src/main/resources/log4j2.xml:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<Configuration status="warn">
+    <Appenders>
+        <Console name="logToConsole" target="SYSTEM_OUT">
+            <PatternLayout pattern="%d [%t] %-5level %logger{36} – %msg%n"/>
+        </Console>
+        <File name="logToFile" fileName="logs/app.log">
+            <PatternLayout pattern="%d [%t] %p %c %m%n"/>
+        </File>
+    </Appenders>
```
This part of configuration file resolves [the problem](https://github.com/mate-academy/jv-logger/pull/1171#pullrequestreview-3351021376).

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: NCM]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

koder95 commented on src/main/resources/log4j2.xml:
```diff
@@ -0,0 +1,21 @@
+            <PatternLayout pattern="%d [%t] %p %c %m%n"/>
+        </File>
+    </Appenders>
+    <Loggers>
+        <Logger name="mate.academy" level="info" additivity="false">
+            <AppenderRef ref="logToConsole"/>
+            <AppenderRef ref="logToFile"/>
+        </Logger>
+        <Root level="error">
+            <AppenderRef ref="logToConsole"/>
+            <AppenderRef ref="logToFile"/>
+        </Root>
+    </Loggers>
```
... and this part too.

koder95 committed:
Reorder variable initialization in Main.java

Moved the initialization of the 'login' variable after the creation of the AuthenticationService instance for improved code readability.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Your solution is approved. Great job replacing all System.out/printStackTrace calls with Log4j2 and using clear, parameterized messages in Main, AuthenticationServiceImpl, and OrderServiceImpl. The .gitignore correctly excludes the logs/ directory. 

Educational override: One requirement appears unverified/missing — a Log4j2 configuration that outputs to both Console and a file under logs/ as required. Without src/main/resources/log4j2.xml, console/file logging isn’t guaranteed. To fix this, add a configuration with a Console appender and a File or RollingFile appender (e.g., fileName="logs/app.log") and reference both in the root logger [CHECKLIST ITEM: logging configuration]. Also, ensure each Java file ends with a trailing newline for consistency.

Keep up the solid work—your logging usage and error handling are on point, and you’re very close to fully meeting all requirements.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: NCM]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: NCM]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

slade13 commented on src/main/java/mate/academy/service/AuthenticationServiceImpl.java:
```diff
@@ -2,15 +2,20 @@
         if (!user.getPassword().equals(password)) {
             throw new AuthenticationException("Username or password are incorrect");
         }
+        log.info("User \"{}\" was logged in", login);
```
You log the username on successful login. Consider masking (***) or hashing in real apps.

koder95 replied:
Really? An administrator should  know who logged in and when for security reasons, shouldn't? Why shouldn't we insert username in log file? I rather insert an IP address and port too.

slade13 reviewed (approved):
Good job!
